Svc/FpySequencer/FpySequencerDirectives.cpp

```diff
@@ -47,4 +86,166 @@ Signal FpySequencer::waitAbs_directiveHandler(const FpySequencer_WaitAbsDirectiv
+
+    if (tlmValue.getBuffLength() == 0) {
+        // could not find this tlm chan
+        // impl of port says to set buf len to 0 in this case
+        return Signal::stmtResponse_failure;
+    }
+
+    Runtime::LocalVariable& lvar = this->m_runtime.localVariables[directive.getdestLvarIndex()];
+    // clear the lvar in case of early return
+    lvar.valueSize = 0;
+    Fw::ExternalSerializeBuffer esb(lvar.value, Fpy::MAX_LOCAL_VARIABLE_BUFFER_SIZE);
+    Fw::SerializeStatus stat = esb.serialize(tlmTime);
+
+    if (stat != Fw::SerializeStatus::FW_SERIALIZE_OK) {
+        // failed to serialize Fw::Time into the lvar
+        return Signal::stmtResponse_failure;
+    }
+
+    lvar.valueSize = esb.getBuffLength();
+    return Signal::stmtResponse_success;
+}
+
+Signal FpySequencer::getTlmValue_directiveHandler(const FpySequencer_GetTlmValueDirective& directive) {
+    if (directive.getdestLvarIndex() >= Fpy::MAX_SEQUENCE_LOCAL_VARIABLES) {
+        return Signal::stmtResponse_failure;
+    }
+    if (!this->isConnected_getTlmChan_OutputPort(0)) {
+        return Signal::stmtResponse_failure;
+    }
+    Fw::Time tlmTime;
+    Fw::TlmBuffer tlmValue;
+    this->getTlmChan_out(0, directive.getchanId(), tlmTime, tlmValue);
+
+    if (tlmValue.getBuffLength() == 0) {
+        // could not find this tlm chan
+        // impl of port says to set buf len to 0 in this case
+        return Signal::stmtResponse_failure;
+    }
+
+    if (tlmValue.getBuffLength() > Fpy::MAX_LOCAL_VARIABLE_BUFFER_SIZE) {
+        // cannot store the tlm value in the lvar
+        return Signal::stmtResponse_failure;
+    }
+    // copy value into lvar
+    Runtime::LocalVariable& lvar = this->m_runtime.localVariables[directive.getdestLvarIndex()];
+    memcpy(lvar.value, tlmValue.getBuffAddr(), tlmValue.getBuffLength());
+    lvar.valueSize = tlmValue.getBuffLength();
+    return Signal::stmtResponse_success;
+}
+
+Signal FpySequencer::getPrmValue_directiveHandler(const FpySequencer_GetPrmValueDirective& directive) {
+    if (directive.getdestLvarIndex() >= Fpy::MAX_SEQUENCE_LOCAL_VARIABLES) {
+        return Signal::stmtResponse_failure;
+    }
+    if (!this->isConnected_prmGet_OutputPort(0)) {
+        return Signal::stmtResponse_failure;
+    }
+    Fw::ParamBuffer prmValue;
+    // set buff len to 0 before call so we can detect if we failed to get it
+    prmValue.setBuffLen(0);
+    Fw::ParamValid valid = this->getParam_out(0, directive.getprmId(), prmValue);
+
+    if (valid != Fw::ParamValid::VALID) {
+        // could not find this prm in the DB
+        return Signal::stmtResponse_failure;
+    }
+
+    if (prmValue.getBuffLength() > Fpy::MAX_LOCAL_VARIABLE_BUFFER_SIZE) {
+        // cannot store the prm value in the lvar
+        return Signal::stmtResponse_failure;
+    }
+    // copy value into lvar
+    Runtime::LocalVariable& lvar = this->m_runtime.localVariables[directive.getdestLvarIndex()];
```

Comment:
Seems like all these copy/sets could be a helper.  This would save very similat code.

```
helper (FwSize index, const U8* buffer, const FwSizeType size) {
   FW_ASSERT buffer != null
    Runtime::LocalVariable& lvar = this->m_runtime.localVariables[index];
     memcpy(lvar.value, buffer,size);
    lvar.valueSize = size;
}
``